.github/workflows/benchmark_execution.yml

```
@@ -67,12 +67,17 @@ jobs:
         run: ./build_tools/scripts/git/update_runtime_submodules.sh
       - name: "Generating benchmark matrix"
         id: generate
-        run: |
-          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}"
-          echo benchmark-matrix=$(jq \
-              'to_entries | map({"device_name": .key, "host_environment": .value.host_environment})' \
-              "${BENCHMARK_CONFIG}") \
-            >> "${GITHUB_OUTPUT}"
+        run: >
+          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}";
+          echo benchmark-matrix=$(jq '[ . | to_entries[]
+            | .key as $device_name
+            | .value.host_environment as $host_environment
+            | (.value.shards | length) as $shard_count
+            | .value.shards[]
+            | { device_name: $device_name, host_environment: $host_environment,
+                shard: { index: .shard_index, count: $shard_count }
+              }
+            ]' "${BENCHMARK_CONFIG}") >> "${GITHUB_OUTPUT}"
```

Taking all my suggestions together

```suggestion
        run: |
          gcloud storage cp "${BENCHMARK_CONFIG_GCS_ARTIFACT}" "${BENCHMARK_CONFIG}"
          echo benchmark-matrix="$(jq -c '[ . | to_entries[]
            | .key as $device_name
            | .value.host_environment as $host_environment
            | (.value.shards | length) as $count
            | .value.shards[]
            | {$device_name, $host_environment, shard: {index, $count}}
            ]' "${BENCHMARK_CONFIG}")" >> "${GITHUB_OUTPUT}"
```

---

Awesome. Thank you for putting all of this into a suggestion